src/notation/internal/notationactioncontroller.cpp

```diff
@@ -540,14 +538,24 @@ bool NotationActionController::canReceiveAction(const ActionCode& code) const
 {
     TRACEFUNC;
 
-    //! NOTE If the notation is not loaded, we cannot process anything.
+    // If no notation is loaded, we cannot handle any action.
     auto masterNotation = currentMasterNotation();
     if (!masterNotation) {
         return false;
     }
 
+    if (code == UNDO_ACTION_CODE) {
+        return canUndo();
+    }
+
+    if (code == REDO_ACTION_CODE) {
+        return canRedo();
+    }
+
+    // Actions other than undo and redo can only be handled when the current
+    // notation contains at least one part.
     if (!masterNotation->hasParts()) {
-        return code == UNDO_ACTION_CODE || code == REDO_ACTION_CODE || code == UNDO_HISTORY_CODE;
+        return false;
```

Comment:
It seems like the comment is no longer relevant with this change.

I was concerned that this change would prevent Undo as well as Redo, so I went to try to test it. But I was thwarted by #25395. Anyway, it's a concern I have that may not be valid.